### DISCUSSION TITLE
default shell commands missing

### DISCUSSION THREAD

**Discussion by tksoh** [eteran/nedit-ng#302] [General]
Nedit 5 comes with a few shell commands by default. But the shell menus on NG is currently looking rather empty. Is this intentional?

Screenshots taken on ubuntu 20.10:

![2021-01-31 12_16_57-toshiba (tksoh's X desktop (tksoh-Satellite-L645_1)) - VNC Viewer](https://user-images.githubusercontent.com/494222/106374497-e6ad8680-63be-11eb-8504-cfcd3cd68b96.jpg)
![2021-01-31 12_17_40-toshiba (tksoh's X desktop (tksoh-Satellite-L645_1)) - VNC Viewer](https://user-images.githubusercontent.com/494222/106374502-e90fe080-63be-11eb-90d3-134c77ee3033.jpg)

**Comment by eteran** [comment]
Hmm, they should be there on Linux for sure. That looks like a bug to me. I'll fix.

**Comment by eteran** [comment]
Strange... I just blew away my `$HOME/.config/nedit-ng` directory and I got the default set just fine...

I also saved and reloaded it. Even added a shell menu entry and saved it, then reloaded.

Does your `$HOME/.config/nedit-ng/shell.yaml` look "normal"?

**Reply by tksoh** [reply]
My shell.yaml is empty somehow. If I remove shell.yaml, then I get the shell menus as expected. I can't totally sure, but I wonder if I might have copy my Windows' config folder into this Ubuntu box to save time having to rebuild everything else on this new Linux box.

Assuming I did copy them from my Windows machine, perhaps this would bring up another questions on the portability of the yaml files. After all, it appears that all the other yaml files are portable between Windows and Linux (Mac?). I am aware the shell commands are highly dependent on the OS and setup.

**Reply by eteran** [reply]
OK, so not sure how or why the file would be empty, but that would explain the empty menus...

As for "portability", yea, the shell menu is the one thing that is going to be decidedly OS specific. It's not even portable between like BSD and Linux because the shell commands have subtle differences.

Maybe post 6.0 we can look into a way to address that in a portable way, not sure what it would look like though.

**Reply by tksoh** [reply]
Definitely something post 6.0. I imaging some kind of flag can be added intp the shell entries to indicate which OS they support, since you seems to already have the info for building NG on different platforms.

**Reply by eteran** [reply]
Right, that's definitely a start. There may even be quirks within a given OS but we'll come up with something!
